AppCenter/AppCenter/Internals/Model/Util/MSAppCenterUserDefaults.m

```diff
@@ -27,13 +27,19 @@ + (instancetype)shared {
   dispatch_once(&onceToken, ^{
     sharedInstance = [[MSAppCenterUserDefaults alloc] init];
     NSDictionary *changedKeys = @{
-      @"MSChannelStartTimer" : @"MSAppCenterChannelStartTimer",         // MSChannelUnitDefault
-      @"pastDevicesKey" : @"MSAppCenterPastDevicesKey",                 // MSDeviceTrackerPrivate
-      @"MSInstallId" : @"MSAppCenterInstallId",                         // MSAppCenterInternal
-      @"MSAppCenterIsEnabled" : @"MSAppCenterAppCenterIsEnabled",       // MSAppCenter
-      @"MSEncryptionKeyMetadata" : @"MSAppCenterEncryptionKeyMetadata", // MSEncrypterPrivate
-      @"SessionIdHistory" : @"MSAppCenterSessionIdHistory",             // MSSessionContext
-      @"UserIdHistory" : @"MSAppCenterUserIdHistory"                    // MSUserIdContext
+      @"MSChannelStartTimer*" : @"MSAppCenterChannelStartTimer",        // [MSChannelUnitDefault oldestPendingLogTimestampKey]
```

Comment:
I am not completely sold with the wildcard thing as it introduces parsing and reserves the `*` character. How about you introduce a new simple type (e.g.:`MSUserDefaultsBaseKey`) inheriting from `NSString`. The new class can simply be declared within `MSAppCenterUserDefaults.h`. It doesn't need to do anything special. Then the migration will just filter on this type.

Reply:
Well, inheriting from `NSString` with `NSString` being a class cluster doesn't make a lot of sense here. Object composition is the right choice indeed.